client/layout/guided-tours/config-elements/step.js

```diff
@@ -150,7 +150,7 @@ export default class Step extends Component {
 
 	wait( props, context ) {
 		if ( isFunction( props.wait ) ) {
-			const ret = props.wait( props, context );
+			const ret = props.wait( { reduxStore: context.store } );
```

Comment:
Doesn't this mean we have to audit all uses of the `wait` prop found in Guided Tours and potentially change their arguments?

Reply:
Oh, okay. Grepping for `wait=` indicates that `checklist-publish-post-tour.js` is the only occurrence.

Reply:
What about passing `context.store` unwrapped here? That would allow consumers to destructure the parameter into e.g. `{ getState, dispatch }` right away -- which to me seems a bit more Redux-y, compared to having to go thru `reduxStore` destructuring (as currently seen below for `checklist-publish-post-tour.js`).

Reply:
> What about passing context.store unwrapped here?

My intent was to leave room for more parameters. Hence an object with named properties. Passing the store "unwrapped" means a growing array of positional arguments, doesn't it?

The client can always destructure with `wait( { reduxStore: { getState, dispatch } } )` if they wish to.

Reply:
> Oh, okay. Grepping for wait= indicates that checklist-publish-post-tour.js is the only occurrence.

Exactly. `wait` is a recent addition that has only one tour using it at the moment.

Reply:
>> What about passing context.store unwrapped here?
>
> My intent was to leave room for more parameters. Hence an object with named properties. 

Yeah, I thought that was the reason. Just comparing it to other Redux stuff (or even `react-redux`' `connect()` -- where there's e.g. `state` and `props`, or `dispatch` and `props` readily available) which seem to be flatter.

> Passing the store "unwrapped" means a growing array of positional arguments, doesn't it?

Do we expect to add many more, except for e.g. `props` (without `store`)?

> The client can always destructure with `wait( { reduxStore: { getState, dispatch } } )` if they wish to.

Yeah, that's a bit ugly tho :smile: 

But anyway, definitely not a blocker.

Reply:
> Yeah, that's a bit ugly tho

I know, but -- the `reduxStore` param will probably be removed soon anyway :) It will be hard to reconcile it with the new React Context API and it can be replaced with the Redux Bridge.